pkg/controller/performanceprofile/components/profileinfo/profileinfo.go

```diff
@@ -0,0 +1,10 @@
+package profileinfo
+
+import performancev2 "github.com/openshift-kni/performance-addon-operators/api/v2"
+
+// PerformanceProfileInfo is a wrapper for PerformanceProfile that can hold extra data and configuration
+type PerformanceProfileInfo struct {
```

Comment:
I am still very skeptical about why do we need this new struct, I do not see a reason to merge this PR until the workload partition API is available and once it will be available you can update PR and we can merge it.
Once you will have API you can fetch the relevant resource under the controller and pass worload partition boolean to the MachineConfig generator.